src/plugins/intel_gna/transformations/insert_reshape_around_matmul.cpp

```diff
@@ -46,6 +47,31 @@ static bool InsertReshape(
     }
 
     auto first_node = iter->second.get_node_shared_ptr();
+    size_t add_input_index = 0;
+    iter = pattern_map.find(add1);
+    std::shared_ptr<ngraph::Node> add_node = nullptr;
+    if (iter != pattern_map.end()) {
+        add_node = iter->second.get_node_shared_ptr();
+        add_input_index = std::dynamic_pointer_cast<ngraph::opset8::MatMul>(add_node->get_input_node_shared_ptr(0)) ? 1 : 0;
+    }
+
+    // If there is an Add layer, check if it doesn't require inserting a Reshape
+    // to align its dimensions with reshaped MatMul's dimensions
+    if (add_node) {
```

Comment:
it can be simplified, we don't need to check here that node is found by pass_matcher, e.g.:
 auto add1_node = pattern_map.at(add1).get_node_shared_ptr();

Reply:
never mind